src/pages/index.js

```diff
@@ -6,74 +6,134 @@ import PopupWithImage from "../components/PopupWithImage.js";
+          cardSection.addItem(cardElement);
+        },
+      },
+      ".gallery__cards"
+    );
+    cardSection.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+api
+  .getUser()
+  .then((inputValues) => {
+    userInfo.setUserInfo(inputValues.name, inputValues.about);
+    userInfo.setAvatar(inputValues.avatar);
+  })
+  .catch((err) => console.error(err));
+
+function handleAddCardFormSubmit(cardData) {
+  newCardPopup.viewLoading(true);
+  api
+    .addCard(cardData)
+    .then((data) => {
+      cardSection.addItem(getCardElement(data));
+      addCardValidator.disableButton();
+      newCardPopup.resetForm();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      newCardPopup.viewLoading(false);
+    });
+}
 
 //----------------------------event handlers---------------------------------//
 function handleImageClick(cardData) {
   popupImage.open(cardData);
 }
 
 function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.job);
-  profileEditValidator.disableButton();
-  profileEditPopup.close();
-}
-
-function handleAddCardFormSubmit(cardData) {
-  cardSection.addItem(
-    getCardElement({ name: cardData.name, link: cardData.link })
-  );
-  addCardValidator.disableButton();
-  newCardPopup.resetForm();
-  newCardPopup.close();
+  profileEditPopup.viewLoading(true);
+  api
+    .editProfile(inputValues.name, inputValues.about)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileEditValidator.disableButton();
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.err(err);
+    })
+    .finally(() => {
+      avatarEditPopup.viewLoading(false);
+    });
 }
 
 //-----------------------------------event listeners--------------------------//
+addNewCardButton.addEventListener("click", () => {
+  newCardPopup.open();
+});
 
 profileEditButton.addEventListener("click", () => {
   const data = userInfo.getUserInfo();
   profileTitleInput.value = data.name;
-  profileDescriptionInput.value = data.job;
+  profileDescriptionInput.value = data.about;
   profileEditPopup.open();
 });
 
-addNewCardButton.addEventListener("click", () => {
-  newCardPopup.open();
-});
-
 //---------------------------------functions---------------------------------//
 function getCardElement(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeIcon
+  );
   const cardElement = card.getView();
   return cardElement;
 }
 
 //-----------------instantiations------------------------------//
 const profileEditValidator = new FormValidator(config, profileEditForm);
 const addCardValidator = new FormValidator(config, addCardFormElement);
+const addAvatarValidator = new FormValidator(config, editAvatarForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```